### DISCUSSION TITLE
Reset on First Day of the Week

### DISCUSSION THREAD

**Discussion by MarVogue** [iSoron/uhabits#975] [Feature Requests]
This is a request for weekly habits: no matter when in the week that an activity is completed, it counts the whole week as a pass. The rest of the days would have the grey checkmarks, even on previous days of that same week. 

However, for simplicity on the concept of "weeks": have the weekly checkmark reset match the First Day of the Week chosen in settings. Other users may enjoy the current weekly setup (you get 1 week after checking off your habit), so it would be best to have this as a toggle option. 

Example: weekly reset on Mondays:
-change settings to First Day of the Week to Mondays
-week now goes from Monday to Sunday
-I check off the activity on Thursday
-the others days in that week (Mon, Tues, Wed, Fri, Sat, Sun) would automatically get the grey checkmark pass. 
-checking something on Sunday (last day) would still give grey checkmarks for all of the previous days up to Monday as it is still in it's week's time.
-checking something on the second Monday would clear the second week and would not affect the first Monday's week. 

And so on for any First Day chosen (I myself start my weeks on Tuesdays haha)

**Comment by ke352802081770314** [comment]
I agree that this method should not replace the current method but would have to be an alternative one. 
It does provide a benefit, because at the moment, weekly means "at least every 7 days" which breaks the series if an activity is performed on different weekdays from one week to the next.
